core/trino-main/src/test/java/io/trino/operator/scalar/FunctionAssertions.java

```diff
@@ -405,6 +404,12 @@ public void assertCachedInstanceHasBoundedRetainedSize(String projection)
         }
     }
 
+    @SuppressWarnings("CheckReturnValue")
+    private void consume(Iterator<Optional<Page>> output)
```

Comment:
It does not "consume", since the iterator is not fully traversed, is it?

Reply:
The `Iterators#getOnlyElement` method will throw an exception if the iterator returns more than one element, so you can look at it two ways: either it does fully iterate because only the happy path matters, or it does not fully iterate because it only takes a one element, even if this is not allowed. I'm inclined to agree with the first interpretation :)